src/SearchParameters/index.js

```diff
@@ -705,16 +705,16 @@ SearchParameters.prototype = {
    * @return {SearchParameters}
    */
   removeNumericRefinement: function(attribute, operator, paramValue) {
-    if (!this.isNumericRefined(attribute, operator)) return this;
-
     if (paramValue !== undefined) {
+      if (!this.isNumericRefined(attribute, operator, paramValue)) return this;
       return this.setQueryParameters({
         page: 0,
         numericRefinements: this._clearNumericRefinements(function(value, key) {
           return key === attribute && value.op === operator && value.val === paramValue;
         })
       });
-    } else if (operator) {
+    } else if (operator !== undefined) {
```

Comment:
It just adds more branches (`if`, `else`, etc) which makes code harder to read. The less code you write, the less bug you have :)